include/video/randominput.h

```diff
@@ -1,40 +1,56 @@
 #pragma once
 
 // BoB robotics includes
-#include "input.h"
+#include "common/stopwatch.h"
+#include "video/input.h"
+
+// Third-party includes
+#include "third_party/units.h"
 
 // OpenCV
 #include <opencv2/opencv.hpp>
 
 // Standard C++ includes
 #include <algorithm>
+#include <chrono>
+#include <limits>
 #include <random>
+#include <thread>
 
 namespace BoBRobotics {
 namespace Video {
 //! A Video::Input which generates white noise, for testing purposes
 template<typename GeneratorType = std::mt19937>
 class RandomInput : public Input {
+    using hertz_t = units::frequency::hertz_t;
+
 public:
-    RandomInput(const cv::Size &size, const std::string &cameraName = "random",
-                const typename GeneratorType::result_type seed = getRandomSeed())
+    RandomInput(const cv::Size &size,
+                typename GeneratorType::result_type seed = std::random_device{}())
       : m_Size(size)
-      , m_CameraName(cameraName)
+      , m_FrameRate(std::numeric_limits<double>::infinity())
+      , m_CameraName("random")
       , m_Generator(seed)
 	  , m_Distribution(0, 0xff)
-    {}
+    {
+        m_FrameTimer.start();
+    }
 
     virtual bool readFrame(cv::Mat &outFrame) override
     {
         outFrame.create(m_Size, CV_8UC3);
         fillRandom(outFrame.data, outFrame.data + m_Size.area() * 3);
+
+        waitForFrameDelay();
```

Comment:
shouldn't these non-sync methods just return false if the stopwatch hasn't reached the delay time? Then the generic method will deal with the waiting for you.

Reply:
True! I wasn't thinking straight